src/llnode_api.cc

```diff
@@ -0,0 +1,226 @@
+#include <lldb/API/SBCommandReturnObject.h>
+#include <lldb/API/SBCompileUnit.h>
+#include <lldb/API/SBDebugger.h>
+#include <lldb/API/SBFileSpec.h>
+#include <lldb/API/SBFrame.h>
+#include <lldb/API/SBModule.h>
+#include <lldb/API/SBProcess.h>
+#include <lldb/API/SBSymbol.h>
+#include <lldb/API/SBTarget.h>
+#include <lldb/API/SBThread.h>
+#include <lldb/lldb-enumerations.h>
+
+#include <algorithm>
+#include <cstring>
+
+#include "src/llnode_api.h"
+#include "src/llscan.h"
+#include "src/llv8.h"
+
+namespace llnode {
+
+LLNodeApi::LLNodeApi()
+    : initialized_(false),
+      debugger(new lldb::SBDebugger()),
+      target(new lldb::SBTarget()),
+      process(new lldb::SBProcess()),
+      llv8(new v8::LLV8()),
+      llscan(new LLScan(llv8.get())) {}
+LLNodeApi::~LLNodeApi() = default;
+LLNodeApi::LLNodeApi(LLNodeApi&&) = default;
+LLNodeApi& LLNodeApi::operator=(LLNodeApi&&) = default;
+
+bool LLNodeApi::debugger_initialized_ = false;
+
+/* Initialize the SB API and load the core dump */
+bool LLNodeApi::Init(const char* filename, const char* executable) {
+  if (!LLNodeApi::debugger_initialized_) {
+    lldb::SBDebugger::Initialize();
+    LLNodeApi::debugger_initialized_ = true;
+  }
+
+  if (initialized_) {
+    return false;
+  }
+
+  *debugger = lldb::SBDebugger::Create();
+  *target = debugger->CreateTarget(executable);
+  if (!target->IsValid()) {
+    return false;
+  }
+
+  *process = target->LoadCore(filename);
+  // Load V8 constants from postmortem data
+  llscan->v8()->Load(*target);
+  initialized_ = true;
+
+  return true;
+}
+
+std::string LLNodeApi::GetProcessInfo() {
+  lldb::SBStream info;
+  process->GetDescription(info);
+  return std::string(info.GetData());
+}
+
+uint32_t LLNodeApi::GetProcessID() { return process->GetProcessID(); }
+
+std::string LLNodeApi::GetProcessState() {
+  int state = process->GetState();
+
+  switch (state) {
```

Comment:
Could we use `SBEvent::GetCStringFromEvent(SBEvent event) -> str const *` instead of the switch?

Reply:
@mmarchini I dug into the headers a little bit and looks like `debugger->StateAsCString` returns what we want too